### DISCUSSION TITLE
/tmp filling up - is this due to Unit?

### DISCUSSION THREAD

**Discussion by jameshulse** [nginx/unit#1004] [Support]
Hi all, we've been running a PHP application on Nginx Unit on an Ubuntu server for a few months. Recently the hard drive filled up and it looks like there are 100's of files belonging to the `unit` user in `/tmp` named like `/tmp/php0aq0sp`.

Is this from Nginx Unit? If so, is it possible to clear these automatically via Unit, or do we need a cron job to do so ourselves.

![image](https://github.com/nginx/unit/assets/797280/d4d13b75-870d-4f46-bd90-f1af388688b6)

Thanks!

**Comment by ac000** [comment]
Hi,

Unit does create temporary files on occasion but does clean them up.

In this case, looking at the naming, they look like they have been created via the PHP [tmpfile()](https://www.php.net/manual/en/function.tmpfile.php) function, but for some reason have not been removed, which can happen when

> The file is automatically removed when closed (for example, by calling [fclose()](https://www.php.net/manual/en/function.fclose.php), or when there are no remaining references to the file handle returned by tmpfile()), or when the script ends.

Unit hasn't been crashing has it?

**Reply by jameshulse** [reply]
Thanks for your reply. Looking at the log there are lots of similar lines about processes starting and ending:

![image](https://github.com/nginx/unit/assets/797280/d98f8467-e6b1-4cab-b20f-575807f88d49)

Every few seconds a new process ends and the application restarts. Maybe this is related to the PHP worker pool settings.

**Comment by ac000** [comment]
Yes, it looks like the application is flapping (but not necessarily crashing, I'd expect to see something about signal 11 or 6 in there if it was).

**Reply by jameshulse** [reply]
The application config in Nginx Unit looks like this. Could the `processes` section possibly be mis-configured and causing the flapping?

```json
{
    ...
    "applications": {
        "laravel": {
            "type": "php 8.2",
            "processes": {
                "max": 20,
                "spare": 5,
                "idle_timeout": 20
            },
            "root": "/var/www/app/public/",
            "script": "index.php"
        }
    }

}

**Comment by ac000** [comment]
Possibly. I'd say an idle timeout of 20 is pretty low, without knowing your workload it's hard to say, but I'd probably go for a couple of minutes or so, allows time for things to settle... is your traffic spiky?

**Reply by jameshulse** [reply]
Thanks. I'll up this to 5 minutes or so and see if that changes things. I would assume it would quickly kill the workers down to the `5` specified and those would sit idle.

Our traffic on this particular environment is very minimal as it is a staging server. In production there is some traffic, but it is an internal tool used a handful of times a day.
